app/src/main/java/com/poupa/vinylmusicplayer/ui/activities/SettingsActivity.java

```diff
@@ -124,6 +141,23 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
 
     public static class SettingsFragment extends ATEPreferenceFragmentCompat implements SharedPreferences.OnSharedPreferenceChangeListener {
 
+        ActivityResultLauncher sharedPreferencesImporter;
+
+        @Override
+        public void onCreate(@Nullable Bundle savedInstanceState) {
+            super.onCreate(savedInstanceState);
+            sharedPreferencesImporter = registerForActivityResult(
```

Comment:
Is this code compatible with older OS ? right now the minimum supported API level is 19

Reply:
You're probably right. I have to check that, probably also at some other new code too.

As I tried out whether this code is compatible with API lvl. 19 I found out that the master branch has issues on an emulator with api lvl 19 following stack trace occured:

App version: 1.8.1-master_e1f089de_240311-2016 CI DEBUG
App version code: 192
Android build version: 4174703
Android release version: 4.4.2
Android SDK version: 19
Android build ID: sdk_x86-eng 4.4.2 KK 4174703 test-keys
Device brand: generic_x86
Device manufacturer: unknown
Device name: generic_x86
Device model: Android SDK built for x86
Device product name: sdk_x86
Device hardware name: ranchu
ABIs: [x86, unknown]
ABIs (32bit): null
ABIs (64bit): null

## Time: 
2024/03/11 19:20:44
## Stack: 
java.lang.NoClassDefFoundError: android.media.AudioAttributes$Builder
	at com.poupa.vinylmusicplayer.service.MusicService.<clinit>(MusicService.java:141)
	at java.lang.Class.newInstanceImpl(Native Method)
	at java.lang.Class.newInstance(Class.java:1208)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:2553)
	at android.app.ActivityThread.access$1800(ActivityThread.java:135)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:136)
	at android.app.ActivityThread.main(ActivityThread.java:5017)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:515)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
	at dalvik.system.NativeStart.main(Native Method)
	
I had to reset my branch on an earlier state cause the device doesn't even open the app anymore without a crash.
Maybe you have an idea or I am just to blind to see a mistake I made.

Reply:
> java.lang.NoClassDefFoundError: android.media.AudioAttributes$Builder

Hmm, I will have a look, thanks

Are you up to date with the master branch? This seems related to an error I introduced and fixed recently.

Reply:
Yes, as you can see in the app version I've checked out the recent master version with the commit-sha: 1.8.1-master_**e1f089de**_240311-2016 CI DEBUG which seems to fit with the recent master commit-sha on gh.

Reply:
Looks like this is my fault. Will revert back to you.

Reply:
Should be fixed with https://github.com/VinylMusicPlayer/VinylMusicPlayer/pull/982 (merged to master)

Reply:
If I verified correctly it should work with sdk lvl 19.